### PR TITLE
feat(pds-ember): GlobalHeader, Avatar, Buttons

### PR DESCRIPTION
**Pull request by CITguy**
## Breaking Changes
* `<Pds::CtaLink>` defaults to a "Secondary" button style (previously "Primary" button styling).
    * This is because it now styles like `<Pds::Button>` in order to support all button styling capabilities.
    * Use `<Pds::CtaLink @variant="primary">` to apply a "Primary" button style.
* Dropped support for button-like `<Pds::Input>` elements.
    * Use `<Pds::Button>` instead of `<Pds::Input @type="button">`
    * Use `<Pds::Button @type="reset">` instead of `<Pds::Input @type="reset">`
    * Use `<Pds::Button @type="submit">` instead of `<Pds::Input @type="submit">`

## Fixes
* `<Pds::Dropdown::Trigger>` will invert the icon when the dropdown is open
* CSS for `<Pds::App::ModalArea>` corrected to prevent blocking `pointer-events`

## Updates
* `<Pds::CtaLink>` now supports all Button styling (e.g., `pds--warning`, `pds--compact`, etc.)
* Add `<Pds::Avatar>` component
* Add `<Pds::GlobalHeader>` component and subcomponents (Logo, PrimaryNav, SecondaryNav, Switcher, UserMenu)
* Add styles to support various dropdown lists (Menu, Nav, Mixed, etc.)
* Refactor CSS to support isolated "theming"

### App
_complete content_
App - shell content complete

### Avatar
_Default_
Avatar - default

_with `@src`_
Avatar - with src

### Button
Button - styles supported on more elements

### CtaLink
CtaLink - supports all Button variants

### Dropdown
_inverted icon when open_
DropdownTrigger - open inverted icon

_Pattern: Nav List_
Dropdown - patterns - Nav List

_Pattern: Menu_
Dropdown - patterns - Menu

_Pattern: Mixed_
Dropdown - patterns - Mixed

### GlobalHeader
GlobalHeader - extra content

_Nav_
GlobalHeader - Nav

_Triggers_
GlobalHeader - Dropdown Trigger

_Dropdowns_
GlobalHeader - Dropdown

## JIRA
* PDS-243
* PDS-245
* PDS-237

### CODE REVIEW

File: packages/pds-ember/addon/components/pds/dropdown/index.hbs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <details
-  class="pds-dropdown {{this.alignmentClass}}"
+  class="pds-dropdown {{this.alignmentClass}} {{@attr-class}}"
```
Comment (CITguy) [inline packages/pds-ember/addon/components/pds/dropdown/index.hbs:2]: This allows me to pass a custom class name to a contextual component like...

```
{{yield (hash 
  Aliased=(component 'pds/dropdown' attr-class='pds-componentName__aliased')
)}}

File: packages/pds-ember/addon/components/pds/avatar/docs.mdx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { ArgsTable } from '@storybook/addon-docs/blocks';
+export const TITLE = 'Components / Avatar';
+
+# Avatar
+
+TBD...
```
Comment (CITguy) [inline packages/pds-ember/addon/components/pds/avatar/docs.mdx:6]: TODO: update docs

File: packages/pds-ember/tests/integration/components/avatar-test.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        template block text
+      </Pds::Avatar>
+    `)
+
+    assert
+      .dom(ROOT)
+      .hasText('template block text')
+
+    // Remove next line and update tests
+    assert.ok(false, 'write additional tests')
```
Comment (CITguy) [inline packages/pds-ember/tests/integration/components/avatar-test.js:40]: write tests

File: packages/pds-ember/tests/integration/components/global-header-test.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        template block text
+      </Pds::GlobalHeader>
+    `)
+
+    assert
+      .dom(ROOT)
+      .hasText('template block text')
+
+    // Remove next line and update tests
+    assert.ok(false, 'write additional tests')
```
Comment (CITguy) [inline packages/pds-ember/tests/integration/components/global-header-test.js:40]: write tests

**Review by CITguy — COMMENTED** [review]
~see comments for changes~ resolved

File: packages/pds-ember/app/styles/pds/core/typography/_config.scss [hunk 5]
```diff
@@ -56,6 +56,7 @@ $_module: "PDS.Core.Typography";
 
 @mixin Interface($size: default) {
   /* [debug] #{$_module}@Interface(#{$size}) */
+  font-family: theme.$fontFamily;
```
[the remark this answers is not in the document]
Comment (CITguy) [reply packages/pds-ember/app/styles/pds/core/typography/_config.scss:59]: yes

File: packages/pds-ember/addon/components/pds/dropdown/docs.mdx [hunk 6]
```diff
@@ -21,12 +21,29 @@ its visibility.
...
-    ...
+    <section>
+      <!-- optional -->
+      <header>...</header>
+
+      <!-- any of the following -->
+      <D.ListItem />
+      <LinkTo />
+      <Pds::Button />
+      <Pds::CtaLink />
```
Comment (meirish) [inline packages/pds-ember/addon/components/pds/dropdown/docs.mdx:32]: This is _great_

File: packages/pds-ember/app/styles/pds/components/button/ghost/index.scss [hunk 7]
```diff
@@ -1,4 +1,4 @@
-@use "config" as *;
+@use "__config" as *;
```
Comment (meirish) [inline packages/pds-ember/app/styles/pds/components/button/ghost/index.scss:1]: nice to see this pattern

File: packages/pds-ember/app/styles/pds/components/dropdown/_header.scss [hunk 8]
```diff
@@ -0,0 +1,16 @@
+///-------------------------------------------------------------------------///
```
Comment (meirish) [inline packages/pds-ember/app/styles/pds/components/dropdown/_header.scss:1]: Should this be `__header.scss` if it's private?

File: packages/pds-ember/app/styles/pds/components/index.scss [hunk 9]
```diff
@@ -1,12 +1,12 @@
 @use "app";
+@use "avatar";
 @use "banner";
 @use "breadcrumbs";
 @use "button";
 @use "button-set";
 @use "checkbox-field";
 @use "code-block";
 @use "copyright";
-@use "cta-link";
```
Comment (meirish) [inline packages/pds-ember/app/styles/pds/components/index.scss:9]: was this removal an accident?

File: packages/pds-ember/app/styles/pds/theme/_vars.scss [hunk 10]
```diff
@@ -114,109 +112,70 @@ $wordSpacing: var(--pds-wordSpacing);
...
-$boxShadow--focus: 0 0 0 3px #{rgba(color.$blue-500, 0.5)};
-// Elevation
-$_black--a05: rgba(0, 0, 0, 0.05);
-$_black--a06: rgba(0, 0, 0, 0.06);
-$_black--a08: rgba(0, 0, 0, 0.08);
-$_black--a10: rgba(0, 0, 0, 0.10);
-$_black--a20: rgba(0, 0, 0, 0.20);
-$_black--a25: rgba(0, 0, 0, 0.25);
+// TODO: refactor/simplify
+$boxShadow--focus: 0 0 0 3px #{$focus-color};
```
Comment (meirish) [inline packages/pds-ember/app/styles/pds/theme/_vars.scss:172]: nice that these are all vars now 👏

**Review by meirish — APPROVED** [review]
Wow this is a monster PR  and it looks great 👏 👍

File: packages/pds-ember/app/styles/pds/components/index.scss (hunk 9 above)
Comment (CITguy) [reply packages/pds-ember/app/styles/pds/components/index.scss:9]: Nope. CtaLink styles are just Button styles.

**Comment by CITguy** [comment]
@meirish : feel free to merge. I no longer have permission to do so.

**Pull request merged**